### PR TITLE
IDESSCD clients

### PR DESCRIPTION
**Pull request by TSchmiedlechner**
In addition to the already existing IPOS clients, we need clients for IDESSD as well.
The following communication protocols were implemented:
- gRPC
- HTTP
- SOAP

Also, integration tests for the respective functionalities were added. As soon as this PR is merged, we can use the new clients in the Launcher and inject them directly into the Middleware.

**Remaining work:**
- [x] Use interface 1.3.1
- [x] Update IDESSCD clients to work with the new package
- [x] Add tests for the new IDESSCD methods
- [x] Remove reference to internal feed when interface 1.3.1 is published (and update package reference)

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="dev" value="https://fiskaltrust.pkgs.visualstudio.com/_packaging/dev/nuget/v3/index.json" />
+  </packageSources>
```
Comment (StefanKert) [inline NuGet.Config:6]: Should we really pull in a dependency to our internal feed?

File: src/fiskaltrust.Middleware.Interface.Client.Http/DESSCD/HttpDeSscd.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+using System.Threading.Tasks;
+
+namespace fiskaltrust.Middleware.Interface.Client.Http
+{
+    internal class HttpDeSscd : IDESSCD
+    {
+        private readonly HttpClient _httpClient;
+        private readonly JsonSerializerSettings _serializerSettings;
+
+        public HttpDeSscd(ClientOptions options)
```
Comment (StefanKert) [inline src/fiskaltrust.Middleware.Interface.Client.Http/DESSCD/HttpDeSscd.cs:16]: HttpDESSCD

File: src/fiskaltrust.Middleware.Interface.Client.Http/DESSCD/HttpDeSscdFactory.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using fiskaltrust.ifPOS.v1.de;
+using fiskaltrust.Middleware.Interface.Client.Common.RetryLogic;
+using System.Threading.Tasks;
+
+namespace fiskaltrust.Middleware.Interface.Client.Http
+{
+    public static class HttpDeSscdFactory
```
Comment (StefanKert) [inline src/fiskaltrust.Middleware.Interface.Client.Http/DESSCD/HttpDeSscdFactory.cs:7]: HttpDESSCDFactory

File: src/fiskaltrust.Middleware.Interface.Client.Soap/SoapDeSscdFactory.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using fiskaltrust.ifPOS.v1.de;
+using fiskaltrust.Middleware.Interface.Client.Common.RetryLogic;
+using System.Threading.Tasks;
+
+namespace fiskaltrust.Middleware.Interface.Client.Soap
+{
+    public static class SoapDeSscdFactory
```
Comment (StefanKert) [inline src/fiskaltrust.Middleware.Interface.Client.Soap/SoapDeSscdFactory.cs:7]: SoapDESSCDFactory

File: src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/DeSscdRetryProxyClient.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        private readonly IRetryPolicyHandler<IDESSCD> _retryPolicyHelper;
+
+        public DeSscdRetryProxyClient(IRetryPolicyHandler<IDESSCD> retryPolicyHelper) => _retryPolicyHelper = retryPolicyHelper;
+
+        public async Task<ScuEchoResponse> EchoAsync(ScuEchoRequest request) => await _retryPolicyHelper.RetryFuncAsync(async (proxy) => await proxy.EchoAsync(request));
+
+        public async Task<EndExportSessionResponse> EndExportSessionAsync(EndExportSessionRequest request) => await _retryPolicyHelper.RetryFuncAsync(async (proxy) => await proxy.EndExportSessionAsync(request));
+
+        public async Task ExecuteSelfTestAsync() => await _retryPolicyHelper.RetryFuncAsync(async (proxy) => await proxy.ExecuteSelfTestAsync());
```
Comment (StefanKert) [inline src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/DeSscdRetryProxyClient.cs:16]: I assume that a void func is working :)

File: src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/RetryPolicyHandler.cs [hunk 6]
```diff
@@ -47,5 +46,35 @@ public async Task<K> RetryFuncAsync<K>(Func<T, Task<K>> action)
...
+
+        public async Task RetryFuncAsync(Func<T, Task> action)
+        {
+            var trial = 0;
+
+            while (trial < _options.Retries)
+            {
+                try
+                {
+                    var timeoutTimer = new Timer(_options.ClientTimeout.TotalMilliseconds);
```
Comment (StefanKert) [inline src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/RetryPolicyHandler.cs:58]: Why do we use a Timer? Can´t we use the CancellationSources?

File: src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/DeSscdRetryProxyClient.cs (hunk 5 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/DeSscdRetryProxyClient.cs:16]: It is, thanks for the reminder :)

File: NuGet.Config (hunk 1 above)
Comment (TSchmiedlechner) [reply NuGet.Config:6]: Good catch. We will need to wait then until the interface 1.3.1-rc2 is on NuGet.org

File: src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/RetryPolicyHandler.cs (hunk 6 above)
Comment (TSchmiedlechner) [reply src/fiskaltrust.Middleware.Interface.Client.Common/RetryLogic/RetryPolicyHandler.cs]: We can and do now :D

**Pull request merged**